lib/redix/uri.ex

```diff
@@ -39,7 +39,7 @@ defmodule Redix.URI do
 
   defp database(%URI{path: "/" <> path = full_path}) do
     case Integer.parse(path) do
-      {db, ""} ->
+      {db, _} ->
```

Comment:
This needs to be

```suggestion
      {db, "/" <> _} ->
```

Otherwise if you have `localhost:6379/0foo` it'll still work and pick up `0`, which is a tad too flexible IMO. Let's go with at least `/0/foo`.

Reply:
This unfortunately doesn't work.

```
# test condition
    opts = opts_from_uri("redis://localhost/2")

# test output...
     ** (ArgumentError) expected database to be an integer, got: "/2"
     code: opts = opts_from_uri("redis://localhost/2")
     stacktrace:
       (redix 1.0.0) lib/redix/uri.ex:46: Redix.URI.database/1
       (redix 1.0.0) lib/redix/uri.ex:16: Redix.URI.opts_from_uri/1
       test/redix/uri_test.exs:42: (test)
```

I could add an additional case like the below to catch the above condition:

```elixir
{db, ""} ->
```

Does that seem reasonable?

Reply:
Oh right sorry I missed that, yes you're right. You can do it in one case

```suggestion
      {db, rest} when rest == "" or binary_part(rest, 0, 1) == "/" ->
```

Reply:
Done!